components/decimal/src/error.rs

```diff
@@ -0,0 +1,13 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Error types for decimal formatting.
+
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("data provider")]
```

Comment:
The examples on https://docs.rs/thiserror/1.0.24/thiserror/ tend to be fairly terse, because I assume they will get interpolated into a more human-readable error message. The more interesting part of the error, if there is one, will be in the details of the data provider error.

Reply:
Okay, actually I do need to add a `{0}` in the display.  Here's what I have now:

```rust
    #[error("error loading data: {0}")]
    Data(#[from] icu_provider::DataError),
```

Sample error message output:

> error loading data: Failed to load resource: Deserializer error: invalid length 3, expected an array of length 10 at line 19 column 3: "/home/sffc/projects/omnicu/resources/testdata/data/json/decimal/symbols@1/bn.json"